### ISSUE TITLE
Apf Missing

### ISSUE THREAD

**Issue by palgunadi1993** [SeisSol/PUMGen#31]
Dear All,
I try to compile the new version of PUMGen, However, I got the following error:
```
-- Cray Programming Environment 2.6.5 C
-- NOTE: LOADEDMODULES changed since initial config!
-- NOTE: this may cause unexpected build errors.
-- Cray Programming Environment 2.6.5 CXX
CMake Warning (dev) at CMakeLists.txt:48 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  Environment variable HDF5_ROOT is set to:

    /opt/cray/pe/hdf5-parallel/1.10.6.1/INTEL/19.1

  For compatibility, CMake is ignoring the variable.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- HDF5: Using hdf5 compiler wrapper for all C compiling
-- Could NOT find APF (missing: APF_INCLUDE_DIR APF_LIBRARIES) 
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
/project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen/APF_INCLUDE_DIR
   used as include directory in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
/project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen/APF_SIM_INCLUDE_DIR
   used as include directory in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
/project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen/APF_ZOLTAN_INCLUDE_DIR
   used as include directory in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
/project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen/GMI_INCLUDE_DIR
   used as include directory in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
/project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen/GMI_SIM_INCLUDE_DIR
   used as include directory in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
/project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen/LION_INCLUDE_DIR
   used as include directory in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
/project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen/MA_INCLUDE_DIR
   used as include directory in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
/project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen/MDS_INCLUDE_DIR
   used as include directory in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
/project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen/MTH_INCLUDE_DIR
   used as include directory in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
/project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen/PCU_INCLUDE_DIR
   used as include directory in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
APF_LIB
    linked by target "pumgen" in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
APF_SIM_LIB
    linked by target "pumgen" in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
APF_ZOLTAN_LIB
    linked by target "pumgen" in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
GMI_LIB
    linked by target "pumgen" in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
GMI_SIM_LIB
    linked by target "pumgen" in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
LION_LIB
    linked by target "pumgen" in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
MA_LIB
    linked by target "pumgen" in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
MDS_LIB
    linked by target "pumgen" in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
MTH_LIB
    linked by target "pumgen" in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen
PCU_LIB
    linked by target "pumgen" in directory /project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen

-- Configuring incomplete, errors occurred!
See also "/project/k1488/kadek/myLibs/PUMGen_SeisSol/PUMGen/build/CMakeFiles/CMakeOutput.log".
make: *** No targets specified and no makefile found.  Stop.
```
Here is my configuration:
```
mkdir build
cd build
/project/k1488/kadek/myLibs/cmakeSeisSol/bin/cmake/bin/cmake .. -DCMAKE_PREFIX_PATH=/project/k1488/kadek/myLibs/PUMGen_SeisSol/core/:/project/k1488/kadek/myLibs/cmakeSeisSol/lib/:/project/k1488/kadek/myLibs/cmakeSeisSol/bin/:/project/k1488/kadek/myLibs/cmakeSeisSol/include/:/project/k1488/kadek/myLibs/cmakeSeisSol/share/ -DSIMMETRIX=ON \
    -DSIMMETRIX_ROOT=/project/k1488/kadek/myLibs/PUMGen_SeisSol/simmodeler/15.0-210501/ -DSIM_MPI=mpich3 \
    -DCMAKE_BUILD_TYPE=Release\
    -DCMAKE_C_COMPILER=cc -DCMAKE_CXX_COMPILER=CC
make -j 24
```
I don't understand with the Apf library. Could you please let me know what is wrong with my configuration?

Thank you for your help.

Best regards,
Kadek

**Comment by Thomas-Ulrich** [comment]
Hi, that's a part of pumi.
See here:
https://github.com/SeisSol/PUMGen/blob/master/cmake/FindApf.cmake
e.g. for me the library is here:
```
di73yeq4@login01:/hppfs/work/pr45fi/di73yeq4/myLibs/core/build/apf_sim> ls
apf_simConfig.h  CMakeFiles  cmake_install.cmake  CTestTestfile.cmake  libapf_sim.a  Makefile
```
maybe double-check: ```-DCMAKE_PREFIX_PATH=$path_to_pumi```

**Comment by palgunadi1993** [comment]
Hi,

I solved the problem. Here is the solution from my side:

1. Recompile pumi with `-DSKIP_SIMMETRIX_VERSION_CHECK=ON` with precompiled Simmodeler 15.0-210501.
2. Separate `-DCMAKE_PREFIX_PATH=$path_to_pumi` that contains the specific library path of HDF5 with `-DHDF5_ROOT=$path_to_HDF5`. 
3. Use cmake > 3.14. In my case, I used cmake 3.16.

Thank you Thomas for responding. I really appreciate it. 

Best regards,
Kadek
